dbt_server/views.py

```diff
@@ -318,19 +366,34 @@ def get_task_status(
 
 
 class PostInvocationRequest(BaseModel):
-    # List of dbt command that will be sent to dbt worker for execution.
-    # E.g. ["--log-format", "json", "run", "--profiles_dir", "testdir"]
+    # Dbt command that will be sent to dbt worker for execution, e.g. [
+    #   "--log-format", "json", "run", "--profiles_dir", "testdir"].
     command: List[str]
-    # Dbt project path, if set --project-dir args will be appended into command
-    # list. If not set, dbt server will fallback to environment variable and
-    # append it. If environment variable is empty, request will be rejected.
-    # Notice user can specify --project-dir args in command directly, in that
-    # case, we will respect user request and won't append anything.
-    project_path: Optional[str]
+    # If set, dbt worker will use it as task_id, otherwise dbt server will
+    # generate a random one and returned. Notice client needs to ensure task_id
+    # uniqueness, post multiple invocations with the same task_id will cause
+    # undetermined behavior.
+    task_id: Optional[str]
+    # Dbt project directory, if set --project-dir args will be appended into
+    # command list. If not set, dbt server will fallback to environment
+    # variable. The process logic is: (top one will override bottom)
```

Comment:
Is there a preference for which the IDE should use? We can do either currently.

Reply:
I suppose you are asking project_dir: I listed couple of pros and cons.

Request project_dir field or Dbt server flags from env var
pros: That's my preference to specify project-dir, either in request or env var is fine(IDE may prefer env var I guess).
cons: We modify user input command(adding --project-dir args) and will IDE expose this in UI? If so is it confusing?

Implicit: task worker flag from env var.
pros: We won't modify user input command.
cons: I foresee it's against future stateless design but I'm 100% sure, because you are configuring dbt worker behavior. In an ideal world IDE should not config anything, everything should come with request body.

Hence my preference is using dbt server flags or project_dir in short term, preferred project_dir in long term after CE takes over pod management.